File: code/java/simplifyPath.java
class Solution {
    public String simplifyPath(String path) {
        Stack<String> temp = new Stack<String>();
        String[] lst = path.split("/");
        
        for(int i=0;i<lst.length;i++){
            if(lst[i].equals("..")==true && temp.isEmpty() == false){
                temp.pop();               
            }
            else if(lst[i].equals(".") == false && lst[i].equals("..") == false && lst[i].equals("") ==false)
                temp.push(lst[i]);
        }
        String res = "";
        while(!temp.isEmpty())
            res = "/" + temp.pop() + res;
        if (res.equals(""))
            return "/";
        return res;
    }
}

File: code/python/simplifyPath.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        if(len(path) == 0):
            return "" 
        lst = path.split("/")
        res = []
        for i in range(len(lst)):
            if(lst[i] == ".." and res != []):
                res.pop(-1)
            else:
                if('.' != lst[i]) and ('..' != lst[i]) and ("" != lst[i]):
                    res.append(lst[i])
        s = ""
        
        for j in range(len(res)):
            s = s + "/" + res[j]
        if s == "":
            return "/"
        return  s
